src/invidious/videos/parser.cr

```diff
@@ -157,7 +158,86 @@ def try_fetch_streaming_data(id : String, client_config : YoutubeAPI::ClientConf
   end
 end
 
-def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any)) : Hash(String, JSON::Any)
+def parse_published_string(item)
```

Comment:
We already have an helper function to do that ^^
Here is the relevant code: https://github.com/iv-org/invidious/blob/069e91d2a609b1df94bdfbb693335969b2a281d8/src/invidious/helpers/utils.cr#L94-L141

Reply:
Thank you!